integrations/unstructured/src/haystack_integrations/components/converters/unstructured/converter.py

```diff
@@ -89,12 +90,21 @@ def to_dict(self) -> Dict[str, Any]:
         )
 
     @component.output_types(documents=List[Document])
-    def run(self, paths: Union[List[str], List[os.PathLike]]):
+    def run(
+        self,
+        paths: Union[List[str], List[os.PathLike]],
+        meta: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]] = None,
+    ):
         """
         Convert files to Haystack Documents using the Unstructured API (hosted or running locally).
 
         :param paths: List of paths to convert. Paths can be files or directories.
             If a path is a directory, all files in the directory are converted. Subdirectories are ignored.
+        :param meta: Optional metadata to attach to the Documents.
+          This value can be either a list of dictionaries or a single dictionary.
+          If it's a single dictionary, its content is added to the metadata of all produced Documents.
+          If it's a list, the length of the list must match the number of sources, because the two lists will be zipped.
```

Comment:
```suggestion
          If it's a list, the length of the list must match the number of paths, because the two lists will be zipped.
          Please note that if the paths contain directories, the length of the meta list must match the actual number of files contained.
```

this should clarify things...